cloud/amazon/ec2_vpc_nacl.py

```diff
@@ -0,0 +1,1472 @@
+       'rule_action': 'allow',
+       'cidr_block': '0.0.0.0/0',
+       'icmp_code': null,
+       'icmp_type': null,
+       'port_range_from': 22,
+       'port_range_to': 22},
+
+      {'rule_number': 200,
+       'protocol': 'tcp',
+       'rule_action': 'allow',
+       'cidr_block': '0.0.0.0/0',
+       'icmp_code': null,
+       'icmp_type': null,
+       'port_range_from': 80,
+       'port_range_to': 80},
+    ]
+    egress: [
+      {'rule_number': 100,
+       'protocol': 'all',
+       'rule_action': 'allow',
+       'cidr_block': '0.0.0.0/0',
+       'icmp_code': null,
+       'icmp_type': null,
+       'port_range_from': null,
+       'port_range_to': null},
+    ]
+    state: 'present'
+
+# Same as above, but with unneeded fields removed (icmp_code, icmp_type)
+- name: "Create and associate production DMZ network ACL"
+  ec2_vpc_nacl:
+    vpc: 'prod'
+    nacl: 'prod-dmz'
+    region: 'us-east-1'
+    subnets: ['prod-dmz-1', 'prod-dmz-2']
+    ingress: [
+      {'rule_number': 100,
+       'protocol': 'tcp',
+       'rule_action': 'allow',
+       'cidr_block': '0.0.0.0/0',
+       'port_range_from': 22,
+       'port_range_to': 22},
+
+      {'rule_number': 200,
+       'protocol': 'tcp',
+       'rule_action': 'allow',
+       'cidr_block': '0.0.0.0/0',
+       'port_range_from': 80,
+       'port_range_to': 80},
+    ]
+    egress: [
+      {'rule_number': 100,
+       'protocol': 'all',
+       'rule_action': 'allow',
+       'cidr_block': '0.0.0.0/0',
+       'port_range_from': null,
+       'port_range_to': null},
+    ]
+    state: 'present'
+
+# Ensure the network ACL named 'legacy-dmz' has no subnets associated with it
+- name: "Disassociate legacy-dmz"
+  ec2_vpc_nacl:
+    vpc: 'core'
+    nacl: 'legacy-dmz'
+    region: 'us-east-1'
+    state: 'disassociated'
+
+# Ensure that a network ACL with a wide-open ingress and egress rule is removed
+- name: "Remove open network ACL"
+  ec2_vpc_nacl:
+      vpc: 'prod'
+      region: 'us-east-1'
+      ingress: [
+          # rule no, protocol, allow/deny, cidr, icmp_code, icmp_type,
+          #                                             port from, port to
+          [100, 'all', 'allow', '0.0.0.0/0', null, null, null, null],
+      ]
+      egress: [
+          [100, 'all', 'allow', '0.0.0.0/0', null, null, null, null]
+      ]
+      state: 'absent'
+
+# Create a rule permitting TCP on all incoming ports on the test VPC
+# (from port 0 to maximum port number)
+- name: "Create NACL permitting TCP on all ports"
+  ec2_vpc_nacl:
+      vpc: 'test'
+      region: 'us-east-1'
+      ingress: [
+          # rule no, protocol, allow/deny, cidr, icmp_code, icmp_type,
+          #                                             port from, port to
+          [100, 'tcp', 'allow', '0.0.0.0/0', null, null, 0, "max"],
+      ]
+      state: 'present'
+
+'''
+
+
+#  imports ####################################################################
+from boto.exception import EC2ResponseError
```

Comment:
for required libs use try/except and set a variable to let user know  in a nice way, otherwise it will generate an exception which is always a bug from a  module. you can look at ec2 module as an example on how to do this.